### PR TITLE
Port to topkg

### PR DESCRIPTION
**Pull request by gjaldon**
Let me know if there's anything I missed. I saw this listed at:
http://canopy.mirage.io/Projects/topkg

### CODE REVIEW

File: opam [hunk 1]
```diff
@@ -8,15 +8,18 @@ dev-repo:     "https://github.com/mirage/ocaml-tuntap.git"
 bug-reports: "https://github.com/mirage/ocaml-tuntap/issues"
 tags:        [ "org:mirage" "org:xapi-project" ]
 
-build:      [make "PREFIX=%{prefix}%"]
-build-test: [make "PREFIX=%{prefix}%" "test"]
-install:    [make "PREFIX=%{prefix}%" "install"]
-remove:     ["ocamlfind" "remove" "tuntap"]
+build:      [ "ocaml" "pkg/pkg.ml" "build" "--pinned" "%{pinned}%" ]
+build-test: [
+  [ "ocaml" "pkg/pkg.ml" "build" "--pinned" "%{pinned}%" ]
```
Comment (samoht) [inline opam]: needs `"--tests" "true"`

Comment (gjaldon) [inline opam]: Just added `"--tests" "true"` in the latest force push.

**Comment by hannesm** [comment]
LGTM!  thanks!

File: pkg/pkg.ml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+#require "topkg"
+open Topkg
+
+
+let () =
+  Pkg.describe "tuntap" @@ fun c ->
+  Ok [
+    Pkg.mllib "lib/tuntap.mllib";
+    Pkg.clib "lib/libtuntap_stubs.clib";
+    Pkg.test "test/getifaddrs_test";
```
Comment (hannesm) [inline pkg/pkg.ml]: I looked around (in nocrypto/ptime/mirage-solo5): you need to add dependencies that `test/*` depends on `tuntap_stubs.o`... this commit https://github.com/hannesm/ocaml-tuntap/commit/ecdfe02374b6898473f7af0eeadf725bc8f61f97 seems to work (although the test _before your topkg port_ do not run smooth on my system (compile and error out); neither do the tests after your topkg port ;)

Comment (hannesm) [inline pkg/pkg.ml]: (clarified comment above)

Comment (gjaldon) [inline pkg/pkg.ml]: Thanks a lot for the commit! Had to read the ocamlbuild manual to understand what was going on. Where did `lib/libtuntap_stubs.a` come from? Is that one of the output files from `Pkg.clib "lib/libtuntap_stubs.clib"`? The `.a` extension is the C equivalent of `.cma` in OCaml? I have minimal knowledge of C and have just been learning it recently.

Btw, can you push your commit into this PR or would you like me to copy the code from there @hannesm?

Comment (hannesm) [inline pkg/pkg.ml]: @gjaldon I don't know how to push that commit here... but I opened #20, which is your commit and mine.

`.a` files are standalone archives, where `.o` files are the object files - this corresponds to `.cmxa` vs `.o` in OCaml (read more about it [here](http://stackoverflow.com/questions/25422560/what-are-all-these-cma-z-files-and-when-we-need-them#25423837) and [here](https://en.wikipedia.org/wiki/Static_library)).

Comment (gjaldon) [inline pkg/pkg.ml]: Thanks a lot for the resources @hannesm! :)

**Comment by gjaldon** [comment]
Closing in favor of PR #20

**Pull request closed without merging**